### ISSUE TITLE
[HOLD for payment 2024-05-02] [$250] Remove `MoneyRequestConfirmationList.js` and copy any changes since Nov 27 into `MoneyTemporaryForRefactorRequestConfirmationList.js`

### ISSUE THREAD

**Issue by tgolen** [Expensify/App#39559]
This is a part of https://github.com/Expensify/App/issues/29107. You can look at that issue for more context behind the cleanup process.

# Problem
The app has two redundant components:

**Old Component**: `MoneyRequestConfirmationList `
**New Component** `MoneyTemporaryForRefactorRequestConfirmationList `

# Solution
Now that most of the referenced components have been cleaned up and following the examples ([example 1](https://github.com/Expensify/App/pull/33646), [example 2](https://github.com/Expensify/App/pull/32764)), the Old Component needs to be completely removed from the codebase

1. Look at the history of the Old Component
1. If there are any changes since Nov 27, 2023 which have not been added to the New Component, copy those changes
1. Replace all uses of the Old Component with the New Component
1. Remove all traces of Old Component
 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a11e5428841cbb31</li>
        <li>Upwork Job ID: 1775636536127291392</li>
        <li>Last Price Increase: 2024-04-03</li>
<li>Automatic offers: </li>
<ul>
<li>eh2077 | Reviewer | 0</li>
<li>brunovjk | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @bfitzexpensify</details>

**Comment by tgolen** [comment]
@bfitzexpensify I might have messed up on the price for this. Can you please confirm for me that the UpWork issue is $250?

**Comment by abzokhattab** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Remove MoneyRequestConfirmationList.js and copy any changes since Nov 27 into IOURequestStepConfirmation.js
### What is the root cause of that problem?
 New Feature 
### What changes do you think we should make in order to solve the problem?
Old Component: `MoneyRequestConfirmationList`
New Component `IOURequestStepConfirmation`



- Look at the history of the Old Component
- If there are any changes since Nov 27, 2023 which have not been added to the New Component, copy those changes
- Replace all uses of the Old Component with the New Component
- Remove all traces of Old Component

**Comment by brunovjk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
This belongs to [Wave 5 cleanup](https://github.com/Expensify/App/issues/29107)
Phase 2 is all about removing the original components and any of the duplicated efforts.

### What is the root cause of that problem?
Replace `MoneyRequestConfirmationList` to `IOURequestStepConfirmation`.

### What changes do you think we should make in order to solve the problem?
> Look at the history of the Old Component
> If there are any changes since Nov 27, 2023 which have not been added to the New Component, copy those changes

1. From [history change](https://github.com/Expensify/App/commits/838b2dd38df388dc028bbde6ebe32997c7fda01d/src/components/MoneyRequestConfirmationList.tsx?since=2023-11-27&until=2024-04-03) I didn't find any changes that should be carried over. Since we have `MoneyTemporaryForRefactorRequestConfirmationList` we were using it and working on it, so all the important changes were made to it.

> Replace all uses of the Old Component with the New Component
> Remove all traces of Old Component

2. We use `MoneyRequestConfirmationList` only at `SplitBillDetailsPage`. We can replace it with `MoneyTemporaryForRefactorRequestConfirmationList` (we already use `MoneyTemporaryForRefactorRequestConfirmationList` in the new component `IOURequestStepConfirmation`).

After remove `MoneyRequestConfirmationList`  rename `MoneyTemporaryForRefactorRequestConfirmationList` and all it uses.

> Be sure to update all routes and navigation to use the new :action param (instead of being hard-coded with "create")
> Update any logic like isEditing to use the new action param from the route

### What alternative solutions did you explore? (Optional)
N/A

**Comment by bfitzexpensify** [comment]
>@bfitzexpensify I might have messed up on the price for this. Can you please confirm for me that the UpWork issue is $250?

Confirmed it's $250

**Comment by eh2077** [comment]
Asked a [question](https://github.com/Expensify/App/issues/29107#issuecomment-2037148188) in the parent issue https://github.com/Expensify/App/issues/29107

**Comment by brunovjk** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/39559#issuecomment-2035695512)

https://github.com/Expensify/App/issues/29107#issuecomment-2037221079: _Remove MoneyRequestConfirmationList.js and copy any changes since Nov 27 into MoneyTemporaryForRefactorRequestConfirmationList.js_

`MoneyRequestConfirmationList` [has been migrated to TypeScript](https://github.com/Expensify/App/pull/37716), so first we need to migrate `MoneyTemporaryForRefactorRequestConfirmationList` to TS as well.

**Comment by eh2077** [comment]
> `MoneyRequestConfirmationList` [has been migrated to TypeScript](https://github.com/Expensify/App/pull/37716), so first we need to migrate `MoneyTemporaryForRefactorRequestConfirmationList` to TS as well.

@brunovjk Thanks for the update. That's a good point. 

@tgolen Should we do migrating `MoneyTemporaryForRefactorRequestConfirmationList` to TS in this task? Or in a separate task just like other TS migration tasks?

**Comment by tgolen** [comment]
Ah, I would probably avoid doing both the TS migration and the cleanup in the same task (to keep the PR scope smaller). Would it be better to do the TS migration first or second?

**Comment by tgolen** [comment]
Sorry everyone, I confused `IOURequestStepConfrimation.js` with `MoneyTemporaryForRefactorRequestConfirmationList.js`. 

This issue doesn't need to have anything major done with `IOURequestStepConfirmation.js`.

This issue should be for replacing `MoneyConfirmationList` with `MoneyTemporaryForRefactorRequestConfirmationList`.

**Comment by abzokhattab** [comment]
Updated [my proposal](https://github.com/Expensify/App/issues/39559#issuecomment-2035632108) according to the last comment

**Comment by brunovjk** [comment]
## Proposal
[Updated](https://github.com/Expensify/App/issues/39559#issuecomment-2035695512)

- Added clarification on passing `splitTag` to `MoneyTemporaryForRefactorRequestConfirmationList`.

**Comment by eh2077** [comment]
> Ah, I would probably avoid doing both the TS migration and the cleanup in the same task (to keep the PR scope smaller). Would it be better to do the TS migration first or second?

Yeah, I agree it's better to do them one by one. I found there's an ongoing PR https://github.com/Expensify/App/pull/37181 to migrate `MoneyTemporaryForRefactorRequestConfirmationList.js`. I think we can wait for that PR.

Once the TS migration [PR](https://github.com/Expensify/App/pull/37181) is merged, I think we can let @brunovjk  to take this task as their [proposal](https://github.com/Expensify/App/issues/39559#issuecomment-2035695512) first pointed out we should replace `MoneyConfirmationList` with `MoneyTemporaryForRefactorRequestConfirmationList` and @brunovjk also started the [discussion](https://github.com/Expensify/App/issues/39559#issuecomment-2037436256) about TS migration.

@tgolen All yours.

🎀👀🎀 C+ reviewed

**Comment by bondydaa** [comment]
I'm going to unassign myself since Tim will be back from OOO tomorrow and it doesn't seem like this is quiet ready to be assigned out? 🤷

**Comment by eh2077** [comment]
@bondydaa Thanks for your comment. This issue needs to wait for the [TS migration](https://github.com/Expensify/App/pull/37181) to be merged, so I think we can wait for @tgolen .

**Comment by brunovjk** [comment]
Thank you, accept the offer. I will wait for the migration to complete to raise the PR. I believe it will be soon.
@eh2077 should we wait the 7 days of the migration regression period to merge our PR?

**Comment by eh2077** [comment]
@brunovjk I think generally you can start working on the PR once the migration PR is merged.

**Comment by brunovjk** [comment]
Thanks :D

**Comment by brunovjk** [comment]
[TS migration](https://github.com/Expensify/App/pull/37181) has been merged. I will start working on our PR immediately.

**Comment by brunovjk** [comment]
[PR](https://github.com/Expensify/App/pull/40176) ready for review

**Comment by brunovjk** [comment]
> LGTM and tested well! https://github.com/Expensify/App/pull/40176#pullrequestreview-2008558539

Yep! We are good to go :D

**Comment by brunovjk** [comment]
We merge @MrMuzyk

**Comment by MrMuzyk** [comment]
@brunovjk I think something might be off when it comes to displaying footer. I can't review the Split expense request
<img width="1920" alt="image" src="https://github.com/Expensify/App/assets/23487612/079bae33-3b31-4649-a6cb-1a6971477ab8">

I might be mistaken but this condition should probably be reversed
https://github.com/Expensify/App/blob/9b2c5187e2203d83bba50bfa66936691788cbd18/src/components/MoneyRequestConfirmationList.tsx#L931

**Comment by brunovjk** [comment]
@MrMuzyk You're right, sorry, I missed that.
What do you think about using the `isReadOnly` variable, something like this:
```
footerContent={!isReadOnly && footerContent}
```
I tested it and it seems to do the job. I can raise a PR to fix today.

**Comment by MrMuzyk** [comment]
At first glance it seems to work 👍 It's probably worth for people who reviewed the PR to have a look at it too.

**Comment by brunovjk** [comment]
>I think something might be off when it comes to displaying footer

@eh2077 Let's fix this before going to QA, let's avoid a regression here :D

**Comment by eh2077** [comment]
@brunovjk Sure. Please tag me when the PR is ready, thank you!

**Comment by eh2077** [comment]
We already have a PR https://github.com/Expensify/App/pull/40659 to fix the DB

**Comment by bfitzexpensify** [comment]
@brunovjk yes, the solution in this issue caused a regression, so the regression penalty applies. 

Payment to @brunovjk complete. 

@eh2077, please complete the BZ checklist so I can finalise payment. Thanks!

**Comment by eh2077** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [x] [@eh2077] The PR that introduced the bug has been identified. Link to the PR: _This is not a bug. It's part of refactoring._
- [x] [@eh2077] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x]  [@eh2077] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] [@eh2077] Determine if we should create a regression test for this bug. N/A

**Comment by eh2077** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [x] [@eh2077] The PR that introduced the bug has been identified. Link to the PR: _This is not a bug. It's part of refactoring._
- [x] [@eh2077] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
- [x]  [@eh2077] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
- [x] [@eh2077] Determine if we should create a regression test for this bug. N/A

**Comment by brunovjk** [comment]
@bfitzexpensify  Could we get an increase in the bounty  here? Mainly to cover the regression. Despite being a simple refactoring here, the component is used in several flows, create and edit as well. Excessive testing was done during and after the PR was merged. Thank you 🙏

**Comment by eh2077** [comment]
To make it easier to evaluate @brunovjk 's request above, I'd like to add on: 

Though we conducted many tests in the main PR https://github.com/Expensify/App/pull/40176, we still slipped up a bit. There're two fix PRs after the main PR 
- https://github.com/Expensify/App/pull/40659
> [!NOTE]
> The first fix PR was up before the main PR hit staging (means the [DB](https://github.com/Expensify/App/issues/40709) was created later) and was merged shortly when the DB was linked, see https://github.com/Expensify/App/issues/39559#issuecomment-2066409074

- https://github.com/Expensify/App/pull/41152
> [!NOTE]
> The second fix PR was a minor issue that reported by https://github.com/Expensify/App/pull/40176#issuecomment-2079801206 (There's no dedicated GH issue created for it)

Below are my two cents on this
Given @brunovjk was dedicated and proactive in whole process of this task 👍, it would be encouraging to be exempt from penalty. Or let's just move on, it happens.

cc @bfitzexpensify @tgolen

**Comment by bfitzexpensify** [comment]
Will defer this one to you @tgolen - what do you think an appropriate bounty is for this issue?

**Comment by tgolen** [comment]
The scope of this issue was pretty large and I am OK authorizing an increase to $500.

**Comment by bfitzexpensify** [comment]
Thanks! 

OK, payment of $500 finalised to @eh2077, and bonus of $375 paid to @brunovjk 

Thanks for the work here, everyone!

**Comment by brunovjk** [comment]
Thank you all! It's always good to contribute to Expensify!!! 🌟
